### PR TITLE
Wellcome to our library

### PR DESCRIPTION
**Pull request by vidalhuix**
## Netlify link
[here.](https://wellcome-to-the-library.netlify.app/)

## Collaborators:
[Looziolooz]

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+.footer-content {
+  color: #fff;
+  background-color: var(--f3);
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0 10px;
+  font-size: .6em;
+}
```
Comment (NellaMori) [inline style.css]: if you want to make the footer sticky, apply these rules to the  `<footer>` element :  
```
footer {
  position: sticky;
  bottom: 0;
  height: 60px;
  color: #fff;
  background-color: var(--f3);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px;
  font-size: .6em;
}
```
like I said in my previous comment, keep either the `footer` or the `<div class="footer-content">` and apply the rules to it ✨

File: style.css [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+body {
+  font-family: Arial, sans-serif;
+  font-size: 100%;
+  margin: 0;
+  padding: 0;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  height: 100%;
```
Comment (NellaMori) [inline style.css:17]: this value should be  `min-height: 100vh;` which applies a value of 100% of the viewport height

File: index.html [hunk 3]
```diff
@@ -3,11 +3,54 @@
...
+      <!-- here is where our books go -->
+    </div>
+  </main>
+  
+  </footer>
+    <div class="footer-content">
+      <p> 2024 Arnau Vidal, Lorenzo Dastoli</p>
+      <img src="books-images/logo.svg" alt="Technigo" class="logo">
+    </div>
+  </footer>
```
Comment (NellaMori) [inline index.html]: Be careful, you're using the footer tag with just closing tags. Also I would remove one of these elements, keep either the ` <footer></footer>` or the ` <div class="footer-content">` here 👍

**Review by NellaMori — CHANGES_REQUESTED** [review]
Hi Arnau & Lorenzo, well done finishing your Library app 📚 ⭐  You followed the instructions and the app fulfills the requirements of filtering and sorting (just for consistency in the app, the sorting by rating should be descending and ascending, both ways). 
You forgot to add the random selection button though, right? It was a basic requirement.
Also make sure the app is responsive for screens of 320px width, all elements should fit and there should not be horizontal scrolling.
I left some comments in the code to improve the UI, add the missing button and you're good to go 🚀

File: index.html (hunk 3 above)
Comment (vidalhuix) [reply index.html]: I saw it while trying to adjust the footer, I just forgot to uppdate it. Good eye!

File: style.css (hunk 1 above)
Comment (vidalhuix) [reply style.css]: I did the changes, and it works fine now😊

**Comment by vidalhuix** [comment]
One bonus thing we did, is that the header image change acoording to the choosed genre :D
The random button is there and working fine. The horizontal on mobile is working too now🎇

File: style.css [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+header {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: center;
+  align-items: center;
```
Comment (ericamechler) [inline style.css]: align-items is mentioned twice. One could be removed :)

File: style.css [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  flex-direction: row;
+  justify-content: space-around;
+  align-items: center;
+  width: 100%;
+  height: 40px;
+  padding: 5px 30px;
+}
+
+button, select, 
+#btnRate, #btnRandom{
```
Comment (ericamechler) [inline style.css:47]: are these really needed? Seems like all of the buttons have the same styling?

File: style.css [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  background-color:rgb(255, 255, 255); 
+  color: var(--f2);
+  border: none;
+  border-radius: 20px;
+  cursor: pointer;
+  height: 100%;
+}
+
+button:hover, select:hover, 
+#btnRandom:hover, #btnRate:hover{
```
Comment (ericamechler) [inline style.css:60]: same comment as above.

File: style.css [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+ padding: 10px;
+ margin-left: 5px;
+}
+
+#az-za {
+  display: flex;
+  justify-content: center;
+}
+
+main, .main{
```
Comment (ericamechler) [inline style.css]: choose one of them?

File: style.css [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+.card div h3{
+  color: var(--f3);
+}
+
+.card div p{
+  color: var(--f2);
+}
+
+/* -----------------------------------------------MEDIA QUERIES--------------------------------------- */
+/* -----------------------------------MOBILE SCREENS 360px to 670xp 1column--------------------------- */
```
Comment (ericamechler) [inline style.css:155]: this is great, very informative comments!

File: style.css [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+
+@media (min-width: 360px) and (max-width:670px){
```
Comment (ericamechler) [inline style.css:157]: around 400px, the buttons stop showing. They are hidden outside of the window for small screens

File: script.js [hunk 10]
```diff
@@ -181,226 +177,110 @@ const books = [
...
-      '2 medium cloves garlic, minced (about 2 teaspoons)',
-      '2 tablespoons vegetable or canola oil',
-      '1 whole flap meat steak, 2 to 2 1/2 pounds',
-      'Kosher salt'
-    ],
-    source: 'Serious Eats',
-    totalTime: 240,
-    url: 'http://www.seriouseats.com/recipes/2012/05/spice-rubbed-grilled-flap-meat-sirloin-tip-recipe.html',
-    image: './recipe-images/grilled.jpg'
+// loadBooks function, this will clear any previous object inside the html container
```
Comment (ericamechler) [inline script.js:192]: easy to understand!

File: script.js [hunk 11]
```diff
@@ -181,226 +177,110 @@ const books = [
...
+const sortBooksByGenre = () => {
+  const value = sortByGenre.value
+  let backgroundImageUrl = "";
+
+  if (value === "all"){
+    loadBooks(BOOKS)
+  } else {
+    const filteredList = BOOKS.filter((book) => book.genre === value)
+
+  switch (value) {
```
Comment (ericamechler) [inline script.js:235]: nice feature!

File: style.css [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+    cursor: pointer;
+  }
+
+  h1{
+    padding: 30px
+  }
+
+}
+/* -----------------------------------TABLET SCREENS 671px to 1024xp 2 colums-------------------------- */
+@media (min-width:671px) and (max-width:1024px) {
```
Comment (ericamechler) [inline style.css:179]: the comment says that there should be 2 columns for the tablet screens, however, that can not be seen

File: style.css [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+
+
+.sortButtons {
+    width: 80%;
+  }
+}
+
+/* -----------------------------------PC SCREENS >1024px 4 columns ------------------------------------- */
```
Comment (ericamechler) [inline style.css]: the comment says that there should be 4 columns, but we can only see 3. Also, maybe the text could be a bit larger?

File: style.css [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+  position: absolute;
+  margin: 10px;
+  font-size: 0.875em;
+  padding: 10px 10px;
+  color: #fff;
+  border-radius: 25px;
+  background-color: var(--f1);
+}
+
+footer {
```
Comment (ericamechler) [inline style.css:124]: the footer text is very small, difficult to read. Maybe make it a bit larger :)

**Review by ericamechler — COMMENTED** [review]
Clean code, easy to understand, especially in the JS-file. Good consistency with the comments

File: style.css (hunk 4 above)
Comment (vidalhuix) [reply style.css]: good eye there, the styling was a little bit like "survival mode coding":)

File: style.css (hunk 5 above)
Comment (vidalhuix) [reply style.css:47]: Good observation:) The buttons share quite the same styling, but some of them have some styling differences that are target whit another selector later in the code. We removed one of the selectors and used just ids instead.
```css
#genreDropdown, 
#btnRate, #btnRandom{
```

File: style.css (hunk 6 above)
Comment (vidalhuix) [reply style.css:60]: to be consistent we changed the code to this:
```css
#az:hover, #za:hover, #genreDropdown:hover, 
#btnRandom:hover, #btnRate:hover{
```

File: style.css (hunk 7 above)
Comment (vidalhuix) [reply style.css]: .main, then

File: style.css (hunk 12 above)
Comment (vidalhuix) [reply style.css:179]: that's true, this was the previuos idea but we forgot to adjust it :)

File: style.css (hunk 13 above)
Comment (vidalhuix) [reply style.css]: the same here.

**Comment by vidalhuix** [comment]
One of the main thing we where proud of, is the hero image changing accordingly to the genre. Go try it:)
Thank's for doing a constructive and educational review.

**Review by NellaMori — APPROVED** [review]

**Pull request closed without merging**